### PR TITLE
Deploy VDDK as side-car container

### PR DESCRIPTION
**Pull request by mareklibra**
The `v2v-vmware` ConfigMap is extended for new `vddk-init-image` key referring image built by cluster user/admin for the [VMware VDDK library](https://www.vmware.com/support/developer/vddk/) based on instructions in Kubevirt's documentation.

The image is used for a side-car container providing the library to the conversion pod,
replacing former vddk-pvc created once per-namespace.

---
- once per cluster, create a `vddk-init` container image, see instructions in https://github.com/oVirt/v2v-conversion-host/commit/bf8bd0a91c3ad67f285c55fe831105c29647db45
- `oc edit configmap v2v-vmware -n kubevirt-hyperconverged`
  - under `data:` section, add new entry based on the first step above:
    ```
    vddk-init-image: your.registry/vddk-init:latest
    ```

Please note, initial version of the `v2v-vmware` ConfigMap in the `kubevirt-hyperconverged` namespace referenced in the instructions above will be created automatically by `v2v operator` which is under development at the time of writing this pull request. It's content can be found i.e. in #532.

### CODE REVIEW

**Comment by mareklibra** [comment]
Cc @suomiy , @nyoxi

File: src/k8s/requests/v2v/constants.js [hunk 1]
```diff
@@ -16,3 +15,6 @@ export const VMWARE_TO_KUBEVIRT_OS_CONFIG_MAP_NAMESPACE = 'kube-public'; // note
 export const VMWARE_TO_KUBEVIRT_OS_CONFIG_MAP_NAME = 'vmware-to-kubevirt-os'; // single configMap per cluster, contains mapping of vmware guestId to common-templates OS ID
 
 export const CONVERSION_SERVICEACCOUNT_DELAY = 2 * 1000; // in ms
+
+export const VMWARE_VDDK_INIT = 'vddk-init';
```
Comment (suomiy) [inline src/k8s/requests/v2v/constants.js]: can we add `CONVERSION_POD` prefix and possible `NAME` suffix?

File: src/k8s/objects/v2v/vmware/conversionPod.js [hunk 2]
```diff
@@ -18,6 +24,20 @@ export const buildConversionPod = ({
...
     serviceAccountName,
+
+    initContainers: [
+      {
+        name: VMWARE_VDDK_INIT,
+        image: vddkInitImage,
+        volumeMounts: [
+          {
+            name: VMWARE_VOLUME_VDDK,
+            mountPath: '/opt/vmware-vix-disklib-distrib',
```
Comment (suomiy) [inline src/k8s/objects/v2v/vmware/conversionPod.js]: can we make a constant out of this?

File: src/k8s/requests/v2v/constants.js (hunk 1 above)
Comment (mareklibra) [reply src/k8s/requests/v2v/constants.js]: sure, done

File: src/k8s/objects/v2v/vmware/conversionPod.js (hunk 2 above)
Comment (mareklibra) [reply src/k8s/objects/v2v/vmware/conversionPod.js]: done

**Comment by mareklibra** [comment]
I just successfully imported a VM with this patch (recent openshift/console/master with this patch applied to web-ui-components).

Prerequisite:
- Have sufficient StorageClass ready and configured for kubevirt (see #528) . The `kubevirt-storage-class-defaults` ConfigMap recently needs to be created manually but there's wip to add it's automatic creation into Kubevirt's deployment.
- Have inital `v2v-vmware` ConfigMap in `kubevirt-hyperconverged` namespace (see #532 , #507). So far this needs to be created manually but there is ongoing work to generate this ConfigMap within kubevirt's deployment by HCO.

Steps to reproduce:
- Build&push vddk-init image (once per cluster, no need to repeat this step for next conversions or changed namespaces):
   - `mkdir myvddk && cd myvddk`
   - downlowad the vddk tar.gz from https://www.vmware.com/support/developer/vddk/
   - `tar -xzf VMware-vix-disklib-6.7.0-8173251.x86_64.tar.gz`   # change name, unpack in  the myvddk directory
   - create Dockerfile with (please note, it's fixed from the version in https://github.com/oVirt/v2v-conversion-host/commit/bf8bd0a91c3ad67f285c55fe831105c29647db45 ):
   ```
   FROM busybox:latest
   COPY vmware-vix-disklib-distrib /vmware-vix-disklib-distrib
   RUN mkdir -p /opt
   ENTRYPOINT ["cp", "-r", "/vmware-vix-disklib-distrib", "/opt"]     
   ```
  - build the image: `docker build -f Dockerfile -t mareklibra/mylib:v670 .`  # adapt based on your account and version
  - push image to a repo: `docker push mareklibra/mylib:v670`
  - warning: it's highly recommended to use a private repository due to VMware's licence
- Let UI know about your new image:
  - `oc edit configmap v2v-vmware -n kubevirt-hyperconverged`
  - and add new entry under `data:` section there: `vddk-init-image: mareklibra/mylib:v670`
- start import in `Create VM Wizard` in the UI
- watch events there or via sort of `oc get pod kubevirt-v2v-conversion-l445b -o yaml`

**Comment by suomiy** [comment]
nice!

**Comment by jwmatthews** [comment]
cc @jmontleon

**Pull request merged**